### ISSUE TITLE
Reason: This Static Web App already has the maximum number of staging environments (3).

### ISSUE THREAD

**Issue by awolad** [Azure/static-web-apps#544]
Copying files to destination directory '/bin/staticsites/ss-oryx/app'...
Done in 6 sec(s).

Removing existing manifest file
Creating a manifest file...
Manifest file created.

Done in 60 sec(s).

---End of Oryx build logs---
Finished building app with Oryx
Found and validated 'routes.json' within build artifacts folder.
Zipping App Artifacts
Done Zipping App Artifacts
Either no Api directory was specified, or the specified directory was not found. Azure Functions will not be created.
The content server has rejected the request with: BadRequest
Reason: This Static Web App already has the maximum number of staging environments (3). Please remove one and try again.

For further information, please visit the Azure Static Web Apps documentation at https://docs.microsoft.com/en-us/azure/static-web-apps/
If you believe this behavior is unexpected, please raise a GitHub issue at https://github.com/azure/static-web-apps/issues/
Exiting

---

My question: is there any way to remove the staging environment automatically?

**Comment by miwebst** [comment]
Hey @awolad, staging environments should get removed automatically when you close a PR. If you need more than 3 open at a time, then you can try the Standard sku. Our quotas can be found here: https://docs.microsoft.com/en-us/azure/static-web-apps/quotas

**Comment by awolad** [comment]
I understand. But it's not removing automatically. Please see my screenshot. 

![image](https://user-images.githubusercontent.com/9510958/129141887-69a7a272-6c68-43eb-874b-47e2eb8f55da.png)

**Here is my workflow yml**

```
name: Azure Static Web Apps CI/CD

on:
  push:
    branches:
      - development
  pull_request:
    types: [opened, synchronize, reopened, closed]
    branches:
      - development

jobs:
  build_and_deploy_job:
    if: github.event_name == 'push' || (github.event_name == 'pull_request' && github.event.action != 'closed')
    runs-on: ubuntu-latest
    name: Build and Deploy Job
    steps:
      - uses: actions/checkout@v2
        with:
          submodules: true
      - name: Build And Deploy
        id: builddeploy
        uses: Azure/static-web-apps-deploy@v1
        with:
          azure_static_web_apps_api_token: ${{ secrets.AZURE_STATIC_WEB_APPS_API_TOKEN_GREEN_BALL }}
          repo_token: ${{ secrets.GITHUB_TOKEN }} # Used for Github integrations (i.e. PR comments)
          action: 'upload'
          ###### Repository/Build Configurations - These values can be configured to match you app requirements. ######
          # For more information regarding Static Web App workflow configurations, please visit: https://aka.ms/swaworkflowconfig
          app_location: '/' # App source code path
          api_location: 'api' # Api source code path - optional
          app_artifact_location: 'dist' # Built app content directory - optional
          app_build_command: 'npm run build:development'
      - uses: actions/checkout@v2
      - name: Use Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v1
        with:
          node-version: ${{ matrix.node-version }}
      # - run: yarn install
      # - run: yarn test
      ###### End of Repository/Build Configurations ######

  close_pull_request_job:
    if: github.event_name == 'pull_request' && github.event.action == 'closed'
    runs-on: ubuntu-latest
    name: Close Pull Request Job
    steps:
      - name: Close Pull Request
        id: closepullrequest
        uses: Azure/static-web-apps-deploy@v1
        with:
          azure_static_web_apps_api_token: ${{ secrets.AZURE_STATIC_WEB_APPS_API_TOKEN_GREEN_BALL }}
          action: 'close'
```

**Comment by awolad** [comment]
It resolved. Thanks @miwebst

**Comment by MohamedN98** [comment]
Hey @awolad. Can you please share the steps you used to resolve this problem. I am having the same issue with Staging Environments not being deleted automatically.
Thanks,
Mohamed

**Comment by awolad** [comment]
Hi @MohamedN98 firstly remove the existing staging environments (if you have) from the Azure portal (have a look at the above screenshot).  And then try again. Make sure you've used the following job for the close pull request. That's it. 

```
  close_pull_request_job:
    if: github.event_name == 'pull_request' && github.event.action == 'closed'
    runs-on: ubuntu-latest
    name: Close Pull Request Job
    steps:
      - name: Close Pull Request
        id: closepullrequest
        uses: Azure/static-web-apps-deploy@v1
        with:
          azure_static_web_apps_api_token: ${{ secrets.AZURE_STATIC_WEB_APPS_API_TOKEN_GREEN_BALL }}
          action: 'close'
```

**Comment by MohamedN98** [comment]
Hey @awolad 
That was it! My YAML workflow was missing the close pull request job.
Initially, I had set up the workflow for Azure Devops by following: https://docs.microsoft.com/en-gb/azure/static-web-apps/build-configuration?tabs=azure-devops#build-and-deploy
That workflow didnt have the close_pull_request_job and when I migrated from Azure Devops to Github Actions, the job was missing from my YAML file.
Thanks for your help!
